svc-bgs-api/src/lib/metric_logger.rb

```diff
@@ -36,8 +36,10 @@ def initialize
     begin
       api_instance = DatadogAPIClient::V1::AuthenticationAPI.new
       api_instance.validate
+    rescue DatadogAPIClient::APIError => e
+      $logger.warn("event=failedDatadogAuthentication error=#{e.class} status=#{e.code} body=#{e.response_body}")
     rescue Exception => e
-      $logger.warn("event=failedDatadogAuthentication error=#{e.message}")
+      $logger.warn("event=failedDatadogAuthentication error=#{e.class}  error=#{e.message}")
```

Comment:
question: is the intent to log a dictionary object of key-values? in which case, will only one of the "error" keys get stored?

Reply:
It depends on the exception that is caught. Most issues with the datadog metrics will be caught in the rescue above. The second one is a catch-all, and we can just log the message. I didn't want to do that with the datadog client because it formats it funky with several newlines. See [here](https://github.com/DataDog/datadog-api-client-ruby/blob/60166bd7d9694515d4daf06c506b7b57f43cdfde/lib/datadog_api_client/api_client.rb#L569) to see how it formats the `e.message`.